dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java

```diff
@@ -519,15 +526,73 @@ private void updateDockerHubImageInformation(List<Tag> newTags, @NotNull Tool to
     private void updateImageInformation(Tool tool, Tag newTag, Tag oldTag) {
         // If old tag does not have image information yet, try to set it. If it does, potentially old tag could have been deleted on
         // GitHub and replaced with tag of the same name. Check that the image is the same. If not, replace.
-        if (oldTag.getImages().isEmpty() && !tool.getRegistry().equals(Registry.DOCKER_HUB.toString())) {
+        if (oldTag.getImages().isEmpty() && tool.getRegistry().equals(Registry.QUAY_IO.toString())) {
             oldTag.getImages().addAll(getImagesForTag(tool, newTag));
-
         } else {
             oldTag.getImages().removeAll(oldTag.getImages());
             oldTag.getImages().addAll(newTag.getImages());
         }
     }
 
+    private List<Tag> getTagsGitLab(Tool tool) {
+        final String repo = tool.getNamespace() + "%2F" + tool.getName();
+        final String projectPath = GITLAB_URL + "projects/" + repo + "/registry/repositories";
+        final List<Tag> tags = new ArrayList<>();
+        Optional<String> projectResponse;
+
+        try {
+            URL projectURL = new URL(projectPath);
+            projectResponse = Optional.of(IOUtils.toString(projectURL, "UTF-8"));
```

Comment:
Can use https://docs.oracle.com/javase/10/docs/api/java/nio/charset/StandardCharsets.html#UTF_8 as a constant for the second parameter